### PR TITLE
Security.md file added

### LINKED ISSUE #12289: Add SECURITY.md
### What would you like to share?

A SECURITY.md file, while not mandatory for all projects, is highly beneficial as it outlines how to report vulnerabilities, enhancing user trust and transparency. It guides contributors in responsible disclosure, aligning with best practices for open-source projects where security is vital. Additionally, many platforms encourage or require it for listing projects in security advisories, and it can define key security policies, such as response times and data handling protocols. Having a `SECURITY.md` is especially valuable for projects involving sensitive data, demonstrating a proactive approach to security management.

### Additional information

I would like to submit a PR if this issue is assigned to me ✅

### PR DESCRIPTION
**Pull request by medss19**
### Describe your change: Fixes #12289

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/arrays/kadanes_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    9
+    >>> kadane.max_subarray_sum([-1, -2, -3, -4])
+    -1
+    >>> kadane.max_subarray_sum([1, 2, 3, 4])
+    10
+    >>> kadane.max_subarray_sum([10, -10, 20, -5, 10])
+    25
+    """
+
+    def __init__(self):
```
[the remark this answers is not in the document]
Comment (medss19) [reply data_structures/arrays/kadanes_algorithm.py:23]: The __init__ method in the KadaneAlgorithm class doesn't return any value, so the correct type hint would be:

`def __init__(self) -> None:`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ repos:
     rev: "v2.5.0"
     hooks:
       - id: pyproject-fmt
+        language_version: python3.12
```
Comment (cclauss) [inline .pre-commit-config.yaml:35]:
```suggestion
        language_version: python3.13
```

File: data_structures/arrays/kadanes_algorithm.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+class KadaneAlgorithm:
```
Comment (cclauss) [inline data_structures/arrays/kadanes_algorithm.py:1]: Please remove as we already have Kadane's algorithm.

File: data_structures/arrays/kadanes_algorithm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cclauss) [reply data_structures/arrays/kadanes_algorithm.py:23]: Then add ` -> None`

**Comment by cclauss** [comment]
I do not think we need a complex security policy.

**Pull request closed without merging**